### PR TITLE
fix: use dht inbound error for decryption (see issue #4596) 

### LINKED ISSUE #4596: [DHT] decrypt should return DhtInboundError

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
Use `DhtInboundError` for DHT message decryption.

Motivation and Context
---
After an internal discussion, it was decided to output an error of `DhtInboundError` instead of `DhtOutboundError`, in `comms/dht/src/crypt.rs:131` and `comms/dht/src/inbound/decryption.rs:409`.

Fixes #4596 

How Has This Been Tested?
---
Run `cargo build`

### CODE REVIEW

File: comms/dht/src/inbound/decryption.rs [hunk 1]
```diff
@@ -68,10 +71,10 @@ enum DecryptionError {
     MessageRejectDecryptionFailed,
     #[error("Failed to decode envelope body")]
     EnvelopeBodyDecodeFailed,
-    #[error("Failed to decrypt message body")]
-    MessageBodyDecryptionFailed,
     #[error("Encrypted message without a destination is invalid")]
     EncryptedMessageNoDestination,
+    #[error("Decryption failed: {0}")]
+    DecryptionFailedMalformedCipher(#[from] DhtInboundError),
```
Comment (sdbondi) [inline comms/dht/src/inbound/decryption.rs]: I think rather use the Decryption error in `crypt.rs`, the inbound and outbound errors are meant to only be used within the inbound and outbound middlewares/pipeline

Comment (sdbondi) [reply comms/dht/src/inbound/decryption.rs]: Let's leave this for now and we can refactor later

**Review by sdbondi — DISMISSED** [review]

**Comment by sdbondi** [comment]
I think let's merge after #4594

**Comment by AaronFeickert** [comment]
> I think let's merge after #4594

Yeah, this will conflict. Plus, there's a desire to provide more specific error types for decryption and padding failure modes in order to make tests cleaner.

**Comment by AaronFeickert** [comment]
Mentioning issue #4596 so it links properly.

**Review by sdbondi — APPROVED** [review]
Perfect - thanks

**Pull request merged**